### ISSUE TITLE
MacOS Sequoia Support

### ISSUE THREAD

**Issue by daylondh** [ev3dev/vscode-ev3dev-browser#121]
**Describe the bug**
Ev3 can be detected, but not connected to, on MacOS Sequoia. This fix required additional OS configuration. See Additional Context below.

**To Reproduce**
Follow normal steps to connect to device. It is visible in the top bar in VSCode. An error with EHOSTUNREACH is listed.

**Expected behavior**
A successful connection as normally seen in prior versions of MacOS, Linux, and Windows.

**Desktop:**
 MacOS Sequoia (15.0+)

**Additional context**
Discovered a fix, which I recommend go into the troubleshooting section in the Wiki. Sequoia has added a "Local Network Access" limitation and VS Code is not automatically included. To address, open Settings->Privacy and Security->Local Network then toggle the slider next to Visual Studio Code to "on".
Another possibility is to see about asking for that permission, but I'm not sure how to do so. As of writing this, very little documentation appears to be available.

**Comment by dlech** [comment]
It looks like this affects lots of other VS Code users for all sorts of things. So hopefully it will get fixed in VS Code. https://github.com/microsoft/vscode/issues/228862

**Comment by daylondh** [comment]
I totally agree, and that's a good reference. Maybe worth mentioning in the Troubleshooting Wiki? I certainly don't think this is an ev3dev browser problem, but it might be helpful.

**Comment by dlech** [comment]
Can you share a screenshot of the settings so I. can put it on the wiki? I haven't updated to Sequoia yet.

**Comment by daylondh** [comment]
Here's a lightly redacted screenshot.

<img width="720" alt="Screenshot 2024-10-08 at 3 40 22 PM" src="https://github.com/user-attachments/assets/904f3542-b774-450d-8a6e-30a79ffa6d58">

**Comment by jdandrea** [comment]
_**UPDATE:** I was able to work around this by switching the VM's Network mode from Bridged to Shared. Once I did that everything "just worked". It's not my preferred network mode but in a pinch it got me out of trouble._

_Also, note that this didn't just affect UTM, it is affecting the latest VMware fusion build as well. While it could be an issue on Apple's end, I do wonder if it's got more to do with apps not all following the security model to a tee, and that Sequoia happened to be the point of no return._

_It reminds me of past major tech cutovers on macOS and other apps that have nothing to do with UTM! In cases like this it can be tough to know and things are not always as they appear. Thoughts welcomed._

x-post: [UTM Issue 6692](https://github.com/utmapp/UTM/issues/6692#issuecomment-2407917881)

----

I upgraded my VS Code and don't see it listed under Privacy & Security > Local Network at all. Even after a reboot and numerous restarts of macOS and VS Code. I'm on Sequoia 15.0.1.

**Comment by dlech** [comment]
Maybe this comment is helpful? Although I would suggest to backup `.vscode` instead of deleting it.

https://github.com/microsoft/vscode/issues/228862#issuecomment-2395509928

**Comment by majnikool** [comment]
In my case, I was able to resolve the issue by ensuring that the VMs were on the same subnet as my laptop (which gets its IP from the router via DHCP). To do this, I changed my router's default DHCP range to match the range of the VMs I had already created. I also used bridge mode in UTM, and everything worked fine with macOS 15.0.1.

If you don't have existing VMs, you can just make sure to create them on the same subnet as your Mac from the start.

**Comment by Optiroc** [comment]
The fix here saved my day after lots of failed troubleshooting and searches for a fix! It should definitely be mentioned in the [troubleshooting document](https://github.com/ev3dev/vscode-ev3dev-browser/wiki/Troubleshooting).
